src/sentry/models/organizationonboardingtask.py

```diff
@@ -71,14 +71,46 @@ def record(self, organization_id, task, **kwargs):
         return False
 
 
-@region_silo_only_model
-class OrganizationOnboardingTask(Model):
+class AbstractOnboardingTask(Model):
     """
-    Onboarding tasks walk new Sentry orgs through basic features of Sentry.
+    An abstract onboarding task that can be subclassed
     """
 
     __include_in_export__ = False
 
+    STATUS_CHOICES = (
+        (OnboardingTaskStatus.COMPLETE, "complete"),
+        (OnboardingTaskStatus.PENDING, "pending"),
+        (OnboardingTaskStatus.SKIPPED, "skipped"),
+    )
+
+    STATUS_KEY_MAP = dict(STATUS_CHOICES)
+    STATUS_LOOKUP_BY_KEY = {v: k for k, v in STATUS_CHOICES}
+
+    organization = FlexibleForeignKey("sentry.Organization")
+    user = FlexibleForeignKey(
+        settings.AUTH_USER_MODEL, null=True, on_delete=models.SET_NULL
+    )  # user that completed
+    status = BoundedPositiveIntegerField(choices=[(k, str(v)) for k, v in STATUS_CHOICES])
+    completion_seen = models.DateTimeField(null=True)
+    date_completed = models.DateTimeField(default=timezone.now)
+    project = FlexibleForeignKey("sentry.Project", db_constraint=False, null=True)
+    data = JSONField()  # INVITE_MEMBER { invited_member: user.id }
+
+    objects = OrganizationOnboardingTaskManager()
+
+    class Meta:
+        abstract = True
+
+    __repr__ = sane_repr("organization", "task")
```

Comment:
@roggenkemper I think this needs to move out of the abstract class and the sandbox version need `user` in there

Reply:
yup, the sandbox version has user here but ill remove this from the abstract version